accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+- Feature Name: Hub Retail Extension
+- Start Date: 2020-11-30
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC describes an extension of retail features for Hub environment.
+
+# Motivation
+[motivation]: #motivation
+
+Reasons for use of Hub with Retail environment
+----------------------------------------------
+
+- scalability
+  - support large environments with the number of terminal exceeding limits of one SUMA server
+
+- disconnected operation
+  - Peripheral server must stay fully functional when the Hub connection is down
+
+
+Possible configurations
+-----------------------
+
+1. Full
+```
+    Terminal -> Branch Server -> Peripheral Server -> Hub
+```
+
+2. Combined SUMA and Branch server 
+```
+    Terminal -> Branch + Peripheral Server Combined -> Hub
+```
+
+3. Standard
+```
+    Terminal -> Branch Server -> SUMA Server
+```
+
+4. Single server
+```
+    Terminal -> Branch + SUMA Server Combined
+```
+
+
+Global Retail data
+------------------
+
+The following items are typically shared in whole organization, so they should be visible and 
+manageable via Hub.
+
+  - Images - distributed as large tarballs via http
+  - Image metadata - pillars, list of packages
+  - HWTYPE configurations - group formula
+  
+Local data
+----------
+
+The following items do not need central management and can stay on Peripheral Server.
+
+  - Branch groups -  groups containing Branch Server and connected terminals
+  - Branch formulas - describe configuration of Branch services - network, pxe, dhcp, dns
+  - Branch PXE data - entries describing how to boot each terminal
+  
+# Detailed design
+[design]: #detailed-design
+
+Images + Image metadata
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
```

Comment:
The image data are too complex, partially stored in files, partially in DB so it needs to go via API to make sure the result is consistent.
ISS then can use this API.

Reply:
New ISS implementation has access to database and file system. We can add any complex logic we need on there. What would be the benefits of having an API endpoint for it instead of using the ISS mechanism?

Reply:
The API gives us more flexibility. I am thinking about the scenario where one Peripheral server will be used in lab, for building and testing images. Only the successfully tested images should be distributed via Hub to other Peripheral servers.
I am not sure if ISS can handle this.

Reply:
In the HUB initial idea, all content should be done in HUB, and then be sync to peripheral servers.
We should be able to build images in HUB, sync them to a lab peripheral server, test them in there. If we customer is satisfied, then they can sync them to all other existing peripheral servers.